src/os/ObjectStore.h

```diff
@@ -1459,6 +1459,9 @@ class ObjectStore {
   virtual void generate_db_histogram(Formatter *f) { }
   virtual void flush_cache() { }
   virtual void dump_perf_counters(Formatter *f) {}
+  virtual int get_cache_obj_count() {
+    return -1;
+  }
```

Comment:
Not too happy about implementing it that way, but wasn't sure if making it a pure method would make things cleaner.

Reply:
You know more about the nitty-gritty of the particular situation than I do, but I'd suggest a pure virtual member function in preference to something that always fails. Always failing /suggests/ that the behavior just isn't valid or available (or belongs in a parent class that may just not exist). That's just my $0.25, and again you know more about the specific situation.

Reply:
RIght, these are good points. This function is just going to disappear, I'm working on a different implementation that takes the Formatter as an argument and prints to it what is appropriate for each object store.